planner/core/logical_plan_builder.go

```diff
@@ -821,6 +821,21 @@ func (b *PlanBuilder) buildSort(p LogicalPlan, byItems []*ast.ByItem, aggMapper
 	sort := LogicalSort{}.Init(b.ctx)
 	exprs := make([]*ByItems, 0, len(byItems))
 	for _, item := range byItems {
+		switch x := item.Expr.(type) {
+		case *driver.ParamMarkerExpr:
+			if b.ctx.GetSessionVars().InPrepare {
+				continue
+			}
+			newItem, isNull, err := expression.ParamToByItemNode(b.ctx, x)
+			if err != nil {
+				err := errors.Errorf("Unknown column '%+v' in 'order clause'", "?")
```

Comment:
use `ErrUnknownColumn.GenWithStackByArgs()`

Reply:
The updated PR should address the comment.